dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -334,6 +334,17 @@ check_no_pkg_source_local() {
         | error_with_message "Don't push PKG_SOURCE_LOCAL definitions upstream"
 }
 
+check_shell_which() {
+    local patterns=()
+    local pathspec=()
+
+    patterns+=(-e '(shell[[:blank:]]\+which')
+
+    pathspec+=('**/Makefile*')
+    git -C "${RIOTBASE}" grep -n "${patterns[@]}" -- "${pathspec[@]}" \
+        | error_with_message "Don't push PKG_SOURCE_LOCAL definitions upstream"
```

Comment:
Wrong error message ;-)

Reply:
In-very-deed. Thanks, fixed. (And tests generally improved, and one for the redirects of #16775 added for good measure.)